api/v1alpha6/conversion_test.go

```diff
@@ -88,6 +88,38 @@ func TestFuzzyConversion(t *testing.T) {
 					status.ExternalNetwork.APIServerLoadBalancer = nil
 				}
 			},
+
+			func(v1alpha8Machine *infrav1.OpenStackMachine, c fuzz.Continue) {
```

Comment:
This is not correct.
@mdbooth found a problem in the conversion functions: `RestoreField` must be provided to a `HashedFieldRestorer`.

I'll wait for him to provide a fix on some other PRs and apply it here.